spec/enumerable_spec.rb

```diff
@@ -0,0 +1,219 @@
+require './my_each.rb'
+
+describe Enumerable do
+  let(:array) { Array.new([1, 2, 3]) }
+  let(:wrong_type) { 'wrong' }
+
+  describe '#my_each' do
+    context 'when applied block of code' do
+      it 'must return the same array' do
+        expect(array.my_each { |x| x + 2 }).to eql(array)
+      end
+
+      it 'must execute for every element' do
+        test_array = []
+        array.my_each { |x| test_array.push(x + 2) }
+        expect(test_array).to eql([3, 4, 5])
+      end
```

Comment:
[Optional]
must execute what? I would say, 'calls the given block for each element'
And you can make this test more solid, by comparing it with the actual `each` method. I just modified your current test into this format,
```ruby
it 'calls the block for each element' do
  test_array = []
  block = proc { |x| test_array.push(x + 2) }
  array.each(&block)
  actual_output = test_array.dup
  test_array = []
  array.my_each(&block)
  expect(test_array).to eql(actual_output)
end
```